javalib/src/main/scala/java/net/URLDecoder.scala

```diff
@@ -0,0 +1,46 @@
+package java.net
+
+import java.io.UnsupportedEncodingException
+import java.nio.{CharBuffer, ByteBuffer}
+import java.nio.charset.Charset
+
+object URLDecoder {
+  def decode(s: String, enc: String): String = {
+
+    if (!Charset.isSupported(enc)) throw new UnsupportedEncodingException(enc)
```

Comment:
Ouch indeed, nice catch.

Well, then you should put the `throw` on the next line, because [ifs and if/elses in statement position should always have their branch(es) on dedicated lines](https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#ifelse)